etc/migration/disable_users_from_tenancy/lib/disable-users-by-tenancy.js

```diff
@@ -0,0 +1,132 @@
+            if (error) {
+                log().error({
+                    'err': err
+                }, 'Unable to delete principals from tenant');
+                process.exit(error.code);
+            }
+
+            log().info("Exiting...");
+            callback(null, users);
+        });
+    });
+};
+
+/**
+ * Page through all the users in the system, and filter them by tenant
+ *
+ * @param  {String}     tenantAlias             The tenant alias to filter by
+ * @param  {Function}   callback                Invoked when users have been collected
+ * @param  {Object}     callback.err            An error that occurred, if any
+ * @param  {Object[]}   callback.userHashes     An array of user hashes
+ * @api private
+ */
+var _getAllUsersForTenant = function (tenantAlias, callback) {
+    var userHashes = [];
+
+    PrincipalsDAO.iterateAll([
+        'principalId', 'tenantAlias', 'email', 'displayName', 'deleted'
+    ], 100, _aggregateUsers, function (err) {
+        if (err) {
+            log().error({
+                'err': err
+            }, 'Failed to iterate all users');
+            return callback(err);
+        }
+
+        log().info('Found %s users for specified tenant', userHashes.length);
+
+        return callback(null, userHashes);
+    });
+
+    /**
+     * Filter users down to those that are part of the specified tenant. Then add them to the `userHashes` array
+     *
+     * @param  {Object[]}   principalHashes       The principals to filter and aggregate
+     * @param  {Function}   callback              Will be invoked when the principals are aggregated
+     */
+    function _aggregateUsers(principalHashes, callback) {
+        log().info('Checking %s principals and filtering...', principalHashes.length);
+        _
+            .chain(principalHashes)
+            .filter(function (principalHash) {
+                return principalHash.tenantAlias === tenantAlias && PrincipalsDAO.isUser(principalHash.principalId);
+            })
+            .each(function (userHash) {
+                log().info('Adding user %s to tenant users', userHash.displayName);
+                userHashes.push(userHash);
+            });
+
+        return callback();
+    }
+};
+
+/**
+ * Deletes users from the database, one by one
+ *
+ * @function _deletePrincipals
+ * @param  {Object[]} usersToDelete Array of users which will be deleted
+ * @param  {Function} afterDeleted  Invoked when all users haven been deleted
+ * @api private
+ */
+function _deletePrincipals(usersToDelete, afterDeleted) {
+    async
+        .map(usersToDelete, function(eachUser, transformed) {
+            PrincipalsDAO.deletePrincipal(eachUser.principalId, function(err) {
+                log().info("Deleted user " + eachUser.displayName + " (" + eachUser.principalId + ")");
+
+                transformed(null, eachUser);
+            });
+
+        }, function (err, results) {
+            afterDeleted(null, results);
+        });
+}
+
+module.exports = {
```

Comment:
`var doMigration = module.exports.doMigration = function(tenantAlias, callback)` would be more consistent with what we're doing across the board.